lib/src/sql/value/value.rs

```diff
@@ -1251,6 +1251,7 @@ pub fn double(i: &str) -> IResult<&str, Value> {
 }
 
 pub fn single(i: &str) -> IResult<&str, Value> {
+	let _diving = crate::sql::parser::depth::dive()?;
```

Comment:
Every recursive codepath I found goes through here, so this is the only place I inserted a call to `dive` ;)